### ISSUE TITLE
Unused links

### ISSUE THREAD

**Issue by estebank** [rust-lang/rust-clippy#15848]
### What it does

Detect docstrings containing unused link definitions.

This might be part of one of the rustdoc related lints, like `clippy::doc_broken_link`.

### Example

```rust
/// Google.
///
/// [Google]: www.google.com
struct G;
```

Should detect that `[Google]` isn't being used.
